gradio/route_utils.py

```diff
@@ -648,41 +650,86 @@ def get_hostname(url: str) -> str:
         return ""
 
 
-class CustomCORSMiddleware(BaseHTTPMiddleware):
-    async def dispatch(self, request: fastapi.Request, call_next):
-        host: str = request.headers.get("host", "")
-        origin: str = request.headers.get("origin", "")
-        host_name = get_hostname(host)
-        origin_name = get_hostname(origin)
+class CustomCORSMiddleware:
+    # This is a simplified and modified version of the Starlette CORSMiddleware that only enables CORS when running on localhost.
```

Comment:
You mean only enables CORS when **not** running on localhost right?

Reply:
I meant when **the host** is running on localhost (that's when we prevent external requests from coming in). Let me clarify